Specify the compression window sizes for zstd-d and br-d

pmeenan opened the pull request:
This change makes it explicit that the `zstd-d` content encoding is `Zstandard with up to a 8 MB compression window` (same as `zstd`) and `br-d` is `Brotli with up to a 16 MB compression window` (same as `br`).

See the discussion in #2754 for why larger ZStandard values were not selected.

This carries some nuance for what encoding to use when compressing for maximum benefit as the resource size crosses beyond 8 MB but that is more of an editorial discussion about the individual compression algorithms and probably better laid out in blog posts or explainers than in the spec.

Fix #2754

pmeenan commented:
@nidhijaju could you PTAL and make sure the language matches what you're working on for the `zstd` update?

martinthomson commented on draft-ietf-httpbis-compression-dictionary.md:
```diff
@@ -345,12 +345,12 @@ Vary: accept-encoding, available-dictionary
+|--------|--------------------------------------------------------------------------------------------------------------------------------|-------------|
+| Name   | Description                                                                                                                    | Reference   |
+|--------|--------------------------------------------------------------------------------------------------------------------------------|-------------|
+| br-d   | A stream of bytes compressed using the Brotli protocol with an external dictionary and a compression window of up to 16 MB.    | {{RFC7932}} |
```
It might be better to avoid tables for this now.

(As a point on style, I find that if you need to define something like this, it is better to add an entire (short) section that includes citations and whatnot.)

martinthomson commented on draft-ietf-httpbis-compression-dictionary.md:
```diff
@@ -300,12 +300,12 @@ negotiating content encoding in HTTP.
-| br-d             | Brotli using an external compression dictionary    |
-| zstd-d           | Zstandard using an external compression dictionary |
-|------------------|----------------------------------------------------|
+|------------------|--------------------------------------------------------------------------------------------|
+| Content-Encoding | Description                                                                                |
+|------------------|--------------------------------------------------------------------------------------------|
+| br-d             | Brotli using an external compression dictionary and a compression window of up to 16 MB.   |
+| zstd-d           | Zstandard using an external compression dictionary and a compression window of up to 8 MB. |
+|------------------|--------------------------------------------------------------------------------------------|
```
A list might be easier to consume here.  This is also the place where you should include citations.  (though you might also include those citations at the first place that Brotli and Zstandard are mentioned in the intro.

pmeenan commented:
Thanks.  I switched the content-encodings definitions and IANA registrations to a list with references back to the algorithm RFCs.  For the IANA registrations I removed the window size from the text and linked it to this document (where the window sizes are defined) instead of to the compression algorithms directly.

Let me know if a more verbose description of the need to specify the window sizes would be useful or if just being explicit about what they are is enough.

nidhijaju reviewed (approved):
lgtm other than the comment
